src/Microsoft.Azure.SignalR.Common/ServiceConnections/ServiceConnectionBase.cs

```diff
@@ -175,11 +185,34 @@ public Task StopAsync()
             catch (Exception ex)
             {
                 Log.UnexpectedExceptionInStop(Logger, ConnectionId, ex);
-            }
-            
+            }
+
             return Task.CompletedTask;
         }
 
+        public async Task CloseAsync()
+        {
+            if (!_closeConnectionLock.Wait(0))
+            {
+                // only one close async process can exist at a time.
+                return;

```

Comment:
line 124 uses `Interlocked.CompareExchange` to make sure `StartAsync` is called only once, do similar to `CloseAsync`?